source/lmp/fix_dplr.cpp

```diff
@@ -148,9 +148,11 @@ void FixDPLR::init() {
   // }
 }
 
+void FixDPLR::setup_pre_force(int vflag) { pre_force(vflag); }
```

Comment:
The purpose of this line is to assign the positions of Wannier centroids at step 0, so that "run 0" can output the correct Wannier centroid positions. 

In LAMMPS, a fix style's setup_pre_force function is executed before setting up and calculating forces, while the setup function of a fix style is executed after  those. We need to add this line to make sure that in step 0, the Wannier centroids are correctly assigned and the electrostatic forces are correctly calculated.

Reply:
Thank you very much for the explanation! 
Then do we need to override the pre_force anymore?

Reply:
Yes, we do, because pre_force is run every step but not automatically run at step 0. Executing pre_force in setup_pre_force corrects the WCs and electrostatic force at step 0.

Reply:
LAMMPS provides documentation: https://docs.lammps.org/Modify_fix.html

Reply:
I see the point. Thank you all!